core/store/migrations/migration_helper.go

```diff
@@ -0,0 +1,417 @@
+package migrations
+
+// NOTE: This package is copied from gormigrate, with code cleanup and applying
+// some fixes around transactional migrations
+// Source: https://github.com/go-gormigrate/gormigrate/blob/dacf763a39d8b491bd13f34bbc583ecb1640094f/go
+//
+// One thing we may wish to do in future is add the InitSchema function which
+// skips migrations and loads the schema from a dump on a virgin database.
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/pkg/errors"
+	"github.com/smartcontractkit/chainlink/core/services/postgres"
+	"go.uber.org/multierr"
+	"gorm.io/gorm"
+)
+
+const (
+	// MaxIndividualMigrationTime is the maximum amount of time a single
+	// migration is allowed to take before declaring it a failure
+	MaxIndividualMigrationTime = 5 * time.Minute
+	initSchemaMigrationID      = "SCHEMA_INIT"
+)
+
+// MigrateFunc is the func signature for migrating.
+type MigrateFunc func(*gorm.DB) error
+
+// RollbackFunc is the func signature for rollbacking.
+type RollbackFunc func(*gorm.DB) error
+
+// InitSchemaFunc is the func signature for initializing the schema.
+type InitSchemaFunc func(*gorm.DB) error
+
+// Options define options for all migrations.
+type Options struct {
+	// TableName is the migration table.
+	TableName string
+	// IDColumnName is the name of column where the migration id will be stored.
+	IDColumnName string
+	// IDColumnSize is the length of the migration id column
+	IDColumnSize int
+	// ValidateUnknownMigrations will cause migrate to fail if there's unknown migration
+	// IDs in the database
+	ValidateUnknownMigrations bool
+}
+
+// Migration represents a database migration (a modification to be made on the database).
+type Migration struct {
+	// ID is the migration identifier. Usually a timestamp like "201601021504".
+	ID string
+	// Migrate is a function that will br executed while running this migration.
+	Migrate MigrateFunc
+	// Rollback will be executed on rollback. Can be nil.
+	Rollback RollbackFunc
+	// DisableDDL disables wrapping the migration in a transaction. Useful in
```

Comment:
nit: Wouldn't just DisableTransaction be more clear? Likewise for the xxxNoDDL - could just xxxNoTransaction or xxxNoTx